packages/react-router/modules/__tests__/matchPath-test.js

```diff
@@ -55,27 +70,37 @@ describe('matchPath', () => {
   })
 
   describe('with no path', () => {
-    it('matches the root URL', () => {
-      const match = matchPath('/test-location/7', {})
-      expect(match).toMatchObject({
+    it('only matches the root URL', () => {
```

Comment:
While a `<Route>` with no path behaves the same, calling `matchPath` behaves differently with no `path` because it now does exact matching by default. The current `<Route>` and `<Switch>` implementations never call `matchPath` for routes with no path, so this shouldn't make any difference.